### DISCUSSION TITLE
For SystemTags docs, list the actual tids with the tag

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6792] [Ideas]
Edit: This ended up as a PR which is now merged #6798 

The discussion can be considered "closed" :-)

---

I would like doc tiddlers about *SystemTags* - for example [this one](https://tiddlywiki.com/prerelease/#SystemTag%3A%20%24%3A%2Ftags%2FPageControls) - to also list the tiddlers with that system tag.

I want this because when creating your own, say, *page control* button, it is natural to copy and modify an existing such tiddler - but it is annoying to not access such system tids easily. (This is actually a general problem.)

To do this, a direct solution would be a conditional viewtemplate (all relevant docs are tagged `SystemTags`) but if this is not liked, then we could for each individual such doc tiddler add the following (because they're all prefixed with `SystemTag: `):

```
<$let tag={{{ [all[current]removeprefix[SystemTag: ]] }}} >

The following tiddlers are tagged with <<tag>>

<$list filter='[all[tiddlers+shadows]tag<tag>sort[]]'>

</$list>
</$let>
```

Actually, it could be made sexier if it was a table also showing caption and icons if there are any.

Thoughts?

**Comment by pmario** [comment]
So every tiddler prefixed `SystemTag: ` should get a new ViewTemplate element that is shown when the tiddler is displayed. right?

**Comment by twMat** [comment]
Whatever method is best to list the relevant tids. I haven't been following the dev very closely so I can't say what the best method is. What would you suggest? :-)

**Comment by twMat** [comment]
Anyone has a suggestions for how to best go about this? 

Conditional viewtemplate?
Separate tid with the template text and then manually add a transclusion to all ~65 tids textfields?
Something-cascades-something?

**Reply by pmario** [reply]
I think it should be done in the same way as for [operator templates](https://tiddlywiki.com/#%24%3A%2Feditions%2Ftw5.com%2Foperator-template) ... but ... with `list-after : $:/core/ui/ViewTemplate/body`  instead of list-before.

**Reply by pmario** [reply]
the code would look similar to

```
<$list filter="[all[current]prefix[SystemTag: ]]">
<$let tag={{{ [all[current]removeprefix[SystemTag: ]] }}} >

The following tiddlers are tagged with <<tag>>

<$list filter='[all[tiddlers+shadows]tag<tag>sort[]]'>

</$list></$let></$list>
```

**Reply by twMat** [reply]
OK, great, so a regular conditional viewtemplate. I can make a PR *but* since I'm not gh compatible, it'll be in the form of an external wiki that I refer to. If you ( @pmario or anyone) could make the PR the proper way that'd be even better.
Here's the complete tiddler. It presents the results in a table so if there is a caption then this is also shown.

```
title: $:/editions/tw5.com/systemtag-template
tags: $:/tags/ViewTemplate
list-after: $:/core/ui/ViewTemplate/body
text:
<$list filter='[all[current]prefix[SystemTag: ]]'>
<$let tag={{{ [all[current]removeprefix[SystemTag: ]] }}} >

<$list filter='[all[tiddlers+shadows]tag<tag>limit[1]]' emptyMessage='(No tiddlers are currently tagged with <<tag>> )'>

The following tiddlers are tagged with <<tag>>

<table>
<tr><th></th><th>caption</th></tr>
<$list filter='[all[tiddlers+shadows]tag<tag>sort[]]'>
<tr>
<td><$link/></td>
<td>{{!!caption}}</td>
</tr>
</$list></table></$list></$let></$list>
```

**Reply by pmario** [reply]
The whole thing is a documentation change and can be done using TiddlyWiki at https://saqimtiaz.github.io/tw5-docs-pr-maker/

The step by step howto is at: https://talk.tiddlywiki.org/t/7-steps-to-improve-the-tiddlywiki-documentation/2590

You can use the main repo to create the PR. Videos are at: https://www.youtube.com/playlist?list=PLuiC_HFhI4Owy3oY_dZpYe7MGlW5JRGIY

**Reply by twMat** [reply]
OK, I tried. It seems to have worked but there was some last step in the form that looked like this

<img width="659" alt="image" src="https://user-images.githubusercontent.com/4534056/179746517-34143544-2358-449f-9c3d-7bd7eba68e81.png">

I have no idea so selected the first option (it was something like "standard tiddler").

If this didn't work, then I will just put up a separate wiki and hope that e.g Jeremy accepts it as a PR. And, frankly, in spite of the heartfelt efforts from you and Saq to simplify the PR process, the next time I have a PR I will not go through these hoops again because it is unfortunately still (way!) to complex and I have to re-watch and re-read everything every time. As you know, I'm a true TW enthusiast and *already a gh contributor* so I can only imagine what the average community member thinks. Github consistently gives me a feeling being stupid. I'm possibly not as smart as the big boys here - but neither is the absolute majority of our community where there are many members who *definitely* could contribute stuff to the project if there was a simple way.

Anyway, I'm off topic but the made PR is hopefully successful (...and accepted! The final hoop - LOL! ) Thank you for your help @pmario
